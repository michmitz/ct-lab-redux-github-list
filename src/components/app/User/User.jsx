import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

const User = () => {
  const userProfile = useSelector(state => state.user);

  return <div data-testid="user-div">
    <h2>{userProfile.name}</h2>
    <h3>Followers: {userProfile.followers}</h3>
    <h3>Following: {userProfile.following}</h3>
    <a href={userProfile.profileLink}>Github Link: {userProfile.profileLink}</a>
  </div>
}

User.propTypes = {
  name: PropTypes.string.isRequired,
  followers: PropTypes.string.isRequired,
  following: PropTypes.string.isRequired,
  profileLink: PropTypes.string.isRequired
}

export default User