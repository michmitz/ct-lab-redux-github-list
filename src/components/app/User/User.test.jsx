import React from 'react';
import { render, cleanup } from '@testing-library/react';
import User from './User';
import { Provider } from 'react-redux';
import store from '../../../store';

describe('User component', () => {
  afterEach(() => cleanup());
  it('renders User', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <User />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});