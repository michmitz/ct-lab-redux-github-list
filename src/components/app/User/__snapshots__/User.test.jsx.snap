// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component renders User 1`] = `
<DocumentFragment>
  <div
    data-testid="user-div"
  >
    <h2 />
    <h3>
      Followers: 
    </h3>
    <h3>
      Following: 
    </h3>
    <a>
      Github Link: 
    </a>
  </div>
</DocumentFragment>
`;
